app/controllers/general.php

```diff
@@ -167,8 +159,22 @@ function router(App $utopia, Database $dbForPlatform, callable $getProjectDB, Sw
             }
         }
 
-        $resourceId = $rule->getAttribute('resourceId');
-        $projectId = $rule->getAttribute('projectId');
+        /** @var Database $dbForProject */
+        $dbForProject = $getProjectDB($project);
+
+        $deployment = Authorization::skip(fn () => $dbForProject->getDocument('deployments', $rule->getAttribute('value')));
+
+        if ($deployment->getAttribute('resourceType', '') === 'functions') {
+            $type = 'function';
+        } elseif ($deployment->getAttribute('resourceType', '') === 'sites') {
+            $type = 'site';
+        }
+
+        $resource = $type === 'function' ?
+            Authorization::skip(fn () => $dbForProject->getDocument('functions', $deployment->getAttribute('resourceId', ''))) :
+            Authorization::skip(fn () => $dbForProject->getDocument('sites', $deployment->getAttribute('resourceId', '')));
```

Comment:
we could already prepare those places with match/switch, that way its prepared for when there is another type added

Reply:
Hmm, true. I copied it from another place 🤔 Im making myself code-quality task to go over entire feat-sites at the end, and replace each of those with match.